docs/StardustDocs/topics/FAQ.md

```diff
@@ -0,0 +1,203 @@
+- Create a `DataFrame` from a `Map` using [`toDataFrame()`](createDataFrame.md#todataframe).
+- Convert a `DataFrame` back to a `Map` using [`toMap()`](toMap.md).
+- Create a [`DataColumn`](DataColumn.md) from a `List` using [`toColumn()`](createColumn.md#tocolumn).
+- Convert a `DataColumn` to a `List` of values.
+- Convert a `DataFrame<T>` into a `List<T>` of data class instances corresponding to each row  
+  using [`toList()`](toList.md).
+
+## Are there any limitations on the types used in a DataFrame?
+
+No! You can store values of **any Kotlin or Java types** inside a [`DataFrame`](DataFrame.md)  
+and work with them in a type-safe manner using [extension properties](extensionPropertiesApi.md)  
+across various [operations](operations.md).
+
+For some commonly used types — such as  
+[Kotlin basic types](https://kotlinlang.org/docs/basic-types.html) and  
+[Kotlin date-time types](https://github.com/Kotlin/kotlinx-datetime) —  
+there is built-in support for automatic conversion and parsing.
+
+## What data sources are supported?
+
+<!------TODO data sources---->
+
+Kotlin DataFrame supports all popular data sources — CSV, JSON, Excel, Apache Arrow, SQL databases, and more!  
+See the [Data Sources section](Data-Sources.md) for a complete list of supported formats  
+and instructions on how to integrate them into your workflow.
+
+Some sources — such as Apache Spark, [Exposed](https://www.jetbrains.com/help/exposed/home.html),  
+and [Multik](https://github.com/Kotlin/multik) — are not supported directly (yet),  
+but you can find [official integration examples here](Integrations.md).
+
+If the data source you need isn't supported yet,  
+feel free to open an [issue](https://github.com/Kotlin/dataframe/issues)  
+and describe your use case — we’d love to hear from you!
+ 
+## I see magically appearing properties in examples. What is it?
+
+These are [extension properties](extensionPropertiesApi.md) — one of the key features of Kotlin DataFrame.
+
+Extension properties correspond to the columns of a [`DataFrame`](DataFrame.md), allowing you to access and select them
+in a **type-safe** and **name-safe** way.
+
+They are generated automatically when working with Kotlin DataFrame in:
+
+- [Kotlin Notebook](gettingStartedKotlinNotebook.md), where extension properties are generated 
+after each cell execution.
+- A Kotlin project in [IntelliJ IDEA](https://www.jetbrains.com/idea/) with the
+  [](Compiler-Plugin.md) enabled, where the properties are generated at compile time.
+
+## I used the KProperties API in older versions, what should I use now that it's deprecated?
+
+The KProperty API was a useful access mechanism in earlier versions.  
+However, with the introduction of [extension properties](extensionPropertiesApi.md) 
+and the [Kotlin DataFrame compiler plugin](Compiler-Plugin.md),
+you now have a more flexible and powerful alternative.
+
+Annotate your Kotlin class with [`@DataSchema`](Compiler-Plugin.md#dataschema-declarations),  
```

Comment:
Or alternatively call `toDataFrame` on list of Kotlin or Java objects, and resulting DataFrame will have schema according to their properties or getters